spec/models/routed_query_spec.rb

```diff
@@ -1,75 +1,127 @@
+            }
           }
-        })
+        )
       end
 
-      it 'should reject the save of the keywords' do
+      let(:expected_errors) do
+        'The following keyword has been duplicated: \'some keyword phrase\'. ' \
+        'Each keyword is case-insensitive and should be added only once.'
+      end
+
+      it 'rejects the save of the keywords' do
         rq = affiliate.routed_queries.build(dup_attributes)
         expect(rq.valid?).to be false
-        expect(rq.errors[:routed_query_keywords]).to include("The following keyword has been duplicated: 'some keyword phrase'. Each keyword is case-insensitive and should be added only once.")
+        expect(rq.errors[:routed_query_keywords]).to include(expected_errors)
       end
     end
 
     context 'when multiple keywords are duplicated' do
       let(:dup_attributes) do
-        valid_attributes.merge({
-          routed_query_keywords_attributes: {
-            '0' => { 'keyword' => 'some keyword phrase' },
-            '1' => { 'keyword' => 'Some Keyword Phrase' },
-            '2' => { 'keyword' => 'some other keyword phrase' },
-            '3' => { 'keyword' => 'Some Other Keyword Phrase' },
+        valid_attributes.merge(
+          {
+            routed_query_keywords_attributes:
+            {
+              '0' => { 'keyword' => 'some keyword phrase' },
+              '1' => { 'keyword' => 'Some Keyword Phrase' },
+              '2' => { 'keyword' => 'some other keyword phrase' },
+              '3' => { 'keyword' => 'Some Other Keyword Phrase' }
+            }
           }
-        })
+        )
+      end
+
+      let(:expected_errors) do
+        'The following keywords have been duplicated: \'some keyword phrase\', ' \
+        '\'some other keyword phrase\'. ' \
+        'Each keyword is case-insensitive and should be added only once.'
       end
 
-      it 'should reject the save of the keywords' do
+      it 'rejects the save of the keywords' do
         rq = affiliate.routed_queries.build(dup_attributes)
         expect(rq.valid?).to be false
-        expect(rq.errors[:routed_query_keywords]).to include("The following keywords have been duplicated: 'some keyword phrase', 'some other keyword phrase'. Each keyword is case-insensitive and should be added only once.")
+        expect(rq.errors[:routed_query_keywords]).to include(expected_errors)
+      end
+    end
+
+    context 'when a keyword contains html' do
+      let(:attributes_with_html) do
+        valid_attributes.merge(
+          {
+            routed_query_keywords_attributes:
+            {
+              '0' => { 'keyword' => 'some <div>keyword</div> phrase' }
+            }
+          }
+        )
+      end
+
+      let(:rq) { affiliate.routed_queries.build(attributes_with_html) }
+
+      before { rq.valid? }
+
+      it 'strips the html from the keyword' do
+        expect(rq.routed_query_keywords[0].keyword).to eq('some keyword phrase')
```

Comment:
These specs should be moved to spec/models/routed_query_keyword_spec.rb. We should test the model instance directly, rather than through an associated object.

Reply:
Moved.